New Resource: azurerm_mariadb_configuration

caius opened the pull request:
Adds a resource to manage configuration against an Azure Database for MariaDB service. Pretty much a shameless copy/paste/amend from the `azure_mysql_configuration` resource.

Sample terraform config:

```tf
resource "azurerm_resource_group" "test" {
  name     = "test"
  location = "North Europe"
}

resource "azurerm_mariadb_server" "test" {
  name                = "test"
  location            = "${azurerm_resource_group.test.location}"
  resource_group_name = "${azurerm_resource_group.test.name}"

  sku {
    name     = "GP_Gen5_2"
    capacity = 2
    tier     = "GeneralPurpose"
    family   = "Gen5"
  }

  storage_profile {
    storage_mb            = 51200
    backup_retention_days = 7
    geo_redundant_backup  = "Disabled"
  }

  administrator_login          = "testlogin"
  administrator_login_password = "testpassword123"
  version                      = "10.2"
  ssl_enforcement              = "Enabled"
}

resource "azurerm_mariadb_configuration" "test_charset" {
  name                = "character_set_server"
  value               = "utf8mb4"
  server_name         = "${azurerm_mariadb_server.test.name}"
  resource_group_name = "${azurerm_resource_group.test.name}"
}
```

katbyte commented on azurerm/resource_arm_mariadb_configuration.go:
```diff
@@ -0,0 +1,155 @@
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+
```
Extra line here
```suggestion
```

katbyte commented on azurerm/resource_arm_mariadb_configuration.go:
```diff
@@ -0,0 +1,155 @@
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
```
Could we validate this with at the very least no zero values?

caius commented:
@katbyte thanks for the feedback, I'd completely overlooked the documentation (duh!). Have added it now. Also addressed the code feedback you'd left, cheers!
